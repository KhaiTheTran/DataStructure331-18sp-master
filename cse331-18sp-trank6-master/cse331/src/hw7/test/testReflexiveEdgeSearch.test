# Test load graph eliminate reflexive node

#Load graph
LoadGraph marveNodeReflexive marveNodeReflexive.tsv

# check graph load
ListNodes marveNodeReflexive
ListChildren marveNodeReflexive x
ListChildren marveNodeReflexive y
ListChildren marveNodeReflexive z

# Test Dijkstra's algorithm shortest path from x to z
FindPath marveNodeReflexive x z

# Test Dijkstra's algorithm shortest path from z to x
FindPath marveNodeReflexive z x

# Test Dijkstra's algorithm shortest path from x to y
FindPath marveNodeReflexive x y

# Test Dijkstra's algorithm shortest path from y to x
FindPath marveNodeReflexive y x

# Test Dijkstra's algorithm shortest path from y to z
FindPath marveNodeReflexive y z

# Test Dijkstra's algorithm shortest path from z to y
FindPath marveNodeReflexive z y