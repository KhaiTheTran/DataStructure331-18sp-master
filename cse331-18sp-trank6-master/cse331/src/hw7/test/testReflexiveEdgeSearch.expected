# Test load graph eliminate reflexive node

#Load graph
loaded graph marveNodeReflexive

# check graph load
marveNodeReflexive contains: x y z
the children of x in marveNodeReflexive are: y1.000
the children of y in marveNodeReflexive are: x1.000 z1.000
the children of z in marveNodeReflexive are: y1.000

# Test Dijkstra's algorithm shortest path from x to z
path from x to z:
x to y with weight 1.000
y to z with weight 1.000
total cost: 2.000

# Test Dijkstra's algorithm shortest path from z to x
path from z to x:
z to y with weight 1.000
y to x with weight 1.000
total cost: 2.000

# Test Dijkstra's algorithm shortest path from x to y
path from x to y:
x to y with weight 1.000
total cost: 1.000

# Test Dijkstra's algorithm shortest path from y to x
path from y to x:
y to x with weight 1.000
total cost: 1.000

# Test Dijkstra's algorithm shortest path from y to z
path from y to z:
y to z with weight 1.000
total cost: 1.000

# Test Dijkstra's algorithm shortest path from z to y
path from z to y:
z to y with weight 1.000
total cost: 1.000